### PR TITLE
updates of palm to gemini and add ask video feature

### PR DESCRIPTION
**Pull request by yudyhendry**

### CODE REVIEW

**Comment by yudyhendry** [comment]
hi @Abhishekbhagwat, PTAL

File: genai-on-vertex-ai/genai-website-mod/backend/config.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [global]
-project_id = ""
-location = ""
+project_id = "kalschi-csm-5"
```
Comment (Abhishekbhagwat) [inline genai-on-vertex-ai/genai-website-mod/backend/config.toml:2]: Let's scrub the project id as its internal only

File: genai-on-vertex-ai/genai-website-mod/backend/config.toml [hunk 2]
```diff
@@ -12,13 +12,13 @@ recs_serving_config_id = ""
 [imagen]
 generation_model_name = "imagegeneration@005"
 caption_model_name = "imagetext@002"
-bucket_name = ""
+bucket_name = "kalschi-for-wsm-images"
```
Comment (Abhishekbhagwat) [inline genai-on-vertex-ai/genai-website-mod/backend/config.toml:15]: Let's remove this bucket names as well and point to the user provided bucket name

File: genai-on-vertex-ai/genai-website-mod/backend/main.py [hunk 3]
```diff
@@ -78,6 +79,29 @@ def get_state_by_action(action: str) -> ContentState:
...
         return ContentState.PRODUCTION
 
+def execute_search_ask_video(thequery: str):
+    """
+    This is a "Ask Video" feature of execute_first_search. As one of two
+    implementation of the web search feature in this app.
+    """
+    request_data = VertexLLMVideoChatRequest(
+        prompt=thequery,
+        video_url="https://www.youtube.com/watch?v=St1up9PwnUY",
```
Comment (Abhishekbhagwat) [inline genai-on-vertex-ai/genai-website-mod/backend/main.py:89]: Is this video url supposed to be hardcoded ?

File: genai-on-vertex-ai/genai-website-mod/backend/routers/vertex_llm_video.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                {
+                    "role": log["role"],
+                    "parts": log["parts"]
+                } for log in req.conversation_logs]
+        except Exception as err:
+            print(f"[Error]Error parsing conversation log:{err}")
+            history = None
+    else:
+        history = None
+        # history=[
```
Comment (Abhishekbhagwat) [inline genai-on-vertex-ai/genai-website-mod/backend/routers/vertex_llm_video.py:59]: Let's remove commented out code

**Review by Abhishekbhagwat — CHANGES_REQUESTED** [review]

File: genai-on-vertex-ai/genai-website-mod/backend/config.toml (hunk 1 above)
Comment (michael-chi) [reply genai-on-vertex-ai/genai-website-mod/backend/config.toml]: removed

File: genai-on-vertex-ai/genai-website-mod/backend/config.toml (hunk 2 above)
Comment (michael-chi) [reply genai-on-vertex-ai/genai-website-mod/backend/config.toml]: removed

File: genai-on-vertex-ai/genai-website-mod/backend/main.py (hunk 3 above)
Comment (michael-chi) [reply genai-on-vertex-ai/genai-website-mod/backend/main.py]: It's intentional.
Move the url to `config.toml` with a default value points to "https://www.youtube.com/watch?v=St1up9PwnUY".

File: genai-on-vertex-ai/genai-website-mod/backend/routers/vertex_llm_video.py (hunk 4 above)
Comment (michael-chi) [reply genai-on-vertex-ai/genai-website-mod/backend/routers/vertex_llm_video.py]: removed

**Review by Abhishekbhagwat — APPROVED** [review]
LGTM

**Pull request merged**